src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -2376,3 +2409,73 @@ def policy_group_exists(self, group_name: str, zonegroup_name: str):
             return True
         except DashboardException:
             return False
+
+
+class RgwTopicmanagement:
+
+    def list_topics(self, uid: Optional[str], tenant: Optional[str]):
+        rgw_topics_list = {}
+        rgw_topic_list_cmd = ['topic', 'list']
+        try:
+            if uid and uid != 'None':
+                rgw_topic_list_cmd.append('--uid')
+                rgw_topic_list_cmd.append(uid)
+
+            elif tenant and tenant != 'None':
+                rgw_topic_list_cmd.append('--tenant')
+                rgw_topic_list_cmd.append(tenant)
+
+            # If both uid and tenant are provided, add both flags and their values
+            elif uid:
+                rgw_topic_list_cmd.append('--uid')
+                rgw_topic_list_cmd.append(uid)
+
+            elif tenant:
+                rgw_topic_list_cmd.append('--tenant')
+                rgw_topic_list_cmd.append(tenant)
+
+            exit_code, out, _ = mgr.send_rgwadmin_command(rgw_topic_list_cmd)
+            if exit_code > 0:
+                raise DashboardException(msg='Unable to fetch topic list',
+                                         http_status_code=500, component='rgw')
+            rgw_topics_list = out
+        except SubprocessError as error:
+            raise DashboardException(error, http_status_code=500, component='rgw')
+        return rgw_topics_list
+
+    def get_topic(self, topic_name: str, tenant: Optional[str]):
+        topic_info = {}
+        rgw_topic_info_cmd = ['topic', 'get']
+        try:
+            if topic_name:
+                rgw_topic_info_cmd.append('--topic', topic_name)
+
+            elif tenant:
```

Comment:
this shouldn't be an elif.
```suggestion
            if tenant:
```

Reply:
changes are done